libimage/load_test.go

```diff
@@ -80,3 +81,78 @@ func TestLoad(t *testing.T) {
 		}
 	}
 }
+
+func TestLoadReference(t *testing.T) {
+	// Make sure that loading images does not leave any artifacts in TMPDIR
+	// behind (containers/podman/issues/14287).
+	tmpdir := t.TempDir()
+	os.Setenv("TMPDIR", tmpdir)
+	defer func() {
+		dir, err := os.ReadDir(tmpdir)
+		require.NoError(t, err)
+		require.Len(t, dir, 0)
```

Comment:
Why is there a test in a defer function? I see no early returns here so this should just be moved at the end of the function IMO

Reply:
Not entirely sure.  Easily fixed, though.